monai/transforms/intensity/array.py

```diff
@@ -1368,8 +1368,25 @@ def __init__(self, num_control_points: Union[Tuple[int, int], int] = 10, prob: f
             if min(num_control_points) <= 2:
                 raise ValueError("num_control_points should be greater than or equal to 3")
             self.num_control_points = (min(num_control_points), max(num_control_points))
-        self.reference_control_points: np.ndarray
-        self.floating_control_points: np.ndarray
+        self.reference_control_points: NdarrayOrTensor
+        self.floating_control_points: NdarrayOrTensor
+
+    def interp(self, x: NdarrayOrTensor, xp: NdarrayOrTensor, fp: NdarrayOrTensor) -> NdarrayOrTensor:
+        ns = torch if isinstance(x, torch.Tensor) else np
+        if isinstance(x, np.ndarray):
+            # approx 2x faster than code below for ndarray
+            return np.interp(x, xp, fp)
+
+        m = (fp[1:] - fp[:-1]) / (xp[1:] - xp[:-1])
+        b = fp[:-1] - (m * xp[:-1])
+
+        indices = ns.searchsorted(xp.reshape(-1), x.reshape(-1), side="left") - 1
```

Comment:
thanks, the solution looks nice, just this function has a compatibility issue with the previous pytorch versions, I think it could be addressed with some version check like this one:
https://github.com/Project-MONAI/MONAI/blob/669bddf581201f994d1bcc0cb780854901605d9b/monai/transforms/spatial/array.py#L232-L234

Reply:
`[torch, numpy].searchsorted(..., side='side')` is the default. So if older torch versions don't have the `side` argument but use `right: bool = False` instead we can simply remove the argument.

Also, for the use case here it doesn't seem relevant: histogram control points with repeated x-values probably don't make much sense.

Reply:
still the previous versions of torch has some issue
```
======================================================================
ERROR: test_interp (tests.test_rand_histogram_shift.TestRandHistogramShift)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/MONAI/MONAI/tests/test_rand_histogram_shift.py", line 60, in test_interp
    yi = tr.interp(array_type([0, 2, 4, 8, 10]), x, y)
  File "/home/runner/work/MONAI/MONAI/monai/transforms/intensity/array.py", line 1387, in interp
    f[x < xp[0]] = fp[0]  # type: ignore
RuntimeError: Index put requires the source and destination dtypes match, got Float for the destination and Long for the source.
----------------------------------------------------------------------

```

btw I suspect the same indices calculation could be done with `torch.bucketize` and maybe it's efficient https://pytorch.org/docs/stable/generated/torch.bucketize.html

Reply:
I think the error is about assigning `fp` (in long) to the larger tensor `f` (in float). perhaps naturally raising this runtime error is not a bad thing to have here...